src/fluree/server/consensus/core.clj

```diff
@@ -153,33 +154,66 @@
     (assoc raft-config :raft-initialized-chan raft-initialized-chan
                        :leader-change-fn leader-change-fn*)))
 
+(defn path-safe-server-address
+  "For creating part of a default log path if not specified in config,
+  we use the multi-address of the server, but we want convert the `/`
+  to `-` so that we don't create nested directories which isn't needed
+  here."
+  [server-address]
+  (-> server-address
+      (str/replace #"^/" "") ;; strip leading /
+      (str/replace #"/$" "") ;; strip trailing /
+      (str/replace "/" "-"))) ;; replace / with -
+
+(defn add-trailing-slash
+  "Adds a '/' to end of string path if it doesn't already exist."
+  [path]
+  (if (str/ends-with? path "/")
+    path
+    (str path "/")))
 
 (defn canonicalize-directories
-  [{:keys [log-directory ledger-directory this-server] :as raft-config}]
-  (let [log-directory*    (-> (or log-directory
-                                  (str "./data/" (name this-server) "/raftlog/"))
-                              io-file/canonicalize-path)
-        ledger-directory* (-> (or ledger-directory
-                                  (str "./data/" (name this-server) "/ledger/"))
-                              io-file/canonicalize-path)]
-    (assoc raft-config :log-directory log-directory*
-                       :ledger-directory* ledger-directory*)))
-
+  [{:keys [log-path conn-storage-path this-server] :as raft-config}]
+  (log/warn "LOG PATH: " log-path)
+  (log/warn "CONN STORAGE PATH: " conn-storage-path)
+  (let [log-path*          (-> (or (some-> log-path add-trailing-slash)
+                                   (str "./data/" (path-safe-server-address this-server) "/_log/"))
+                               io-file/canonicalize-path)
+        conn-storage-path* (-> (or conn-storage-path
+                                   (str "./data/" (path-safe-server-address this-server) "/"))
+                               io-file/canonicalize-path)]
+    (log/warn "LOG PATH*: " log-path*)
+    (log/warn "CONN STORAGE PATH*: " conn-storage-path*)
+    (assoc raft-config :log-path log-path*
+                       :log-directory log-path* ;; note raft library config is `:log-directory`
+                       :conn-storage-path conn-storage-path*)))
+
+
+(defn is-shared-storage?
+  "If conn method is :file or :memory, returns false, else true."
+  [conn]
+  (let [method (conn-proto/-method conn)]
+    (if (#{:file :memory} method)
```

Comment:
Can we reference this set from a shared ns? I see it's defined as `fluree.server.components.consensus/replicate-file-conn-types`, but not sure if we can access that here w/o a circular ref.